### PR TITLE
Fix compilation of fullblock closures

### PR DESCRIPTION
**Pull request by tesonep**
Taking all the changes I have done to fix some of the problems.

@MarcusDenker 
@StevenCostiou 
@guillep

### CODE REVIEW

File: src/Kernel/CompiledCode.class.st [hunk 1]
```diff
@@ -571,6 +586,12 @@ CompiledCode >> primitiveErrorVariableName [
 	^nil
 ]
 
+{ #category : #testing }
+CompiledCode >> propertyAt: aString ifAbsent: aFullBlockClosure [ 
+	
+	^ aFullBlockClosure value
```
Comment (jecisc) [inline src/Kernel/CompiledCode.class.st:592]: Is thins for polymorphism with something else?

I find it strange to have the concept of properties and just evaluate the if absent block.

Comment (tesonep) [reply src/Kernel/CompiledCode.class.st:592]: Yes, the compiled methods have properties but the blocks do not.

**Comment by Ducasse** [comment]
Ouch not build passes.

**Comment by Ducasse** [comment]
None of the builds built.

and we have + ./pharo Pharo.image test --junit-xml-output --stage-name=Tests-osx-32 '.*'
2019-09-04 14:10:08.641 Pharo[21139:507] ApplePersistenceIgnoreState: Existing state will not be touched. New state will be written to /var/folders/w0/nn_htp7n6ddcydvkv21hjzhc0000gn/T/org.pharo.Pharo.savedState
2019-09-04 14:10:08.892 Pharo[21139:507] 14:10:08.892 ERROR:     [0xa146e1a8] AQMEIO.cpp:377: _FindIOUnit: error -66680
2019-09-04 14:10:08.892 Pharo[21139:507] 14:10:08.892 ERROR:     [0xa146e1a8] >aq> AudioQueueObject.cpp:1590: Prime: failed (-66680); will stop (5512/0 frames)
2019-09-04 14:10:08.893 Pharo[21139:507] 14:10:08.893 ERROR:     [0xa146e1a8] AQMEIO.cpp:377: _FindIOUnit: error -66680
2019-09-04 14:10:08.893 Pharo[21139:507] 14:10:08.893 ERROR:     [0xa146e1a8] >aq> AudioQueueObject.cpp:1590: Prime: failed (-66680); will stop (33075/0 frames)

**Comment by MarcusDenker** [comment]
There is a conflict, so the build builds the old version without the fix that was merged yesterday

**Comment by Ducasse** [comment]
ouch 99 broken tests, so what do we do?

**Comment by MarcusDenker** [comment]
a lot of them are the decompiler. The decompiler does not yet support fullblock closures.

**Comment by Ducasse** [comment]
So how do we proceed?
Do we have a doc, comment, description of the new blockclosure?

**Comment by MarcusDenker** [comment]
I will first make a list of all failing tests.  (I will run the tests  a couple of times, some seem to not be related)

I already saw that some tests in MethodMapTest are failing, I will look at those next

**Comment by MarcusDenker** [comment]
The crash on linux might be due to a not up-to-date implementation of #rfEnsure:, I did a PR to fix it: https://github.com/pharo-project/pharo/issues/4542 

(when it is integrated I will close/reopen this PR so we build with the fix included)

**Comment by MarcusDenker** [comment]
There are more problems with this method... I will continue to debug.

**Comment by Ducasse** [comment]
It would be nice to see if we can pair program because I would like to learn the decompiler.

**Comment by MarcusDenker** [comment]
Yes, let's plan do so some pair programming from next week on on this

**Comment by Ducasse** [comment]
I cannot wednesday afternoon (CST thomas)and thursday full afternoon

**Comment by MarcusDenker** [comment]
The Reflectivity related crash is now fixed. This is a cleaned up list of the failing test:

FFI:

testCqsort – Unix32.UnifiedFFI.Tests.Tests.FFICallbackTest
testCqsortWithByteArray – Unix32.UnifiedFFI.Tests.Tests.FFICallbackTest


Debugger:

testInterruptedContext – Unix32.Debugger.Tests.DebugSessionContexts2Test
testContextsAfterStepInto – Unix32.Debugger.Tests.DebugSessionContexts2Test
testStepThrough – Unix32.Debugger.Tests.StepThroughTest


Decompiler: 
	- FBDDecompilerTest
	- OCBytecodeDecompilerExamplesTest

Fuel: 
	- FLBlockClosureSerializationTest



Opal (Mapping):

testExampleTempNamedTempVectorNestedBlock – Unix32.OpalCompiler.Tests.Misc.MethodMapTest
testSourceMappingBlock – Unix32.OpalCompiler.Tests.Misc.MethodMapTest
testDNU – Unix32.OpalCompiler.Tests.Misc.MustBeBooleanTest




Misc:

testAllExamples – Unix32.Hiedra.Tests.Model.HiExamplesTest
testTerminatingBlockedCriticalSectionShouldNotUnblockAnotherWaitingSection – 
Unix32.Kernel.Tests.Extended.Processes.MutexTest
testClosureRestart – Unix32.Kernel.Tests.Methods.ContextTest
testMixedMethod – Unix32.Kernel.Tests.Processes.LocalRecursionStopperTest
testPointersTo – Unix32.Kernel.Tests.Objects.ProtoObjectTest
testPointersToCycle – Unix32.Kernel.Tests.Objects.ProtoObjectTest
testMixedMethod – Unix32.Kernel.Tests.Processes.RecursionStopperTest
testInCriticalWait – Unix32.Kernel.Tests.Processes.SemaphoreTest
testMethodsEnumeration – Unix32.Calypso.SystemPlugins.Traits.Queries.Tests.ClyTraitUserScopeTest
testMethodsEnumerationWhenBothMetaLevels – Unix32.Calypso.SystemPlugins.Traits.Queries.Tests.ClyTraitUserScopeTest

**Comment by MarcusDenker** [comment]
I suggest to do these things:

1) commit all but the setup change. I will make a new PR for that.

2) do the switch to the Sista bytemode backend first without new blocks.

3) then we make a new PR for testing the change to new blocks

**Comment by Ducasse** [comment]
Gogogo

**Comment by MarcusDenker** [comment]
I have made a PR with all the image level changes: https://github.com/pharo-project/pharo/pull/5076

I will close this and open a new one for actually testing making it default.

**Pull request closed without merging**